x/restaking/simulation/msg_factory.go

```diff
@@ -0,0 +1,613 @@
+		weightMsgRemoveOperatorhFromAllowList = DefaultWeightMsgRemoveOperatorFromAlloList
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgBorrowPoolSecurity, &weightMsgBorrowPoolSecurity, nil, func(_ *rand.Rand) {
+		weightMsgBorrowPoolSecurity = DefaultWeightMsgBorrowPoolSecurity
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgCeasePoolSecurityBorrow, &weightMsgCeasePoolSecurityBorrow, nil, func(_ *rand.Rand) {
+		weightMsgCeasePoolSecurityBorrow = DefaultWeightMsgCeasePoolSecurityBorrow
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgDelegatePool, &weightMsgDelegatePool, nil, func(_ *rand.Rand) {
+		weightMsgDelegatePool = DefaultWeightMsgDelegatePool
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgDelegateOperator, &weightMsgDelegateOperator, nil, func(_ *rand.Rand) {
+		weightMsgDelegateOperator = DefaultWeightMsgDelegateOperator
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgDelegateService, &weightMsgDelegateService, nil, func(_ *rand.Rand) {
+		weightMsgDelegateService = DefaultWeightMsgDelegateService
+	})
+
+	appParams.GetOrGenerate(OperationWeightMsgSetUserPreferences, &weightMsgSetUserPreferences, nil, func(_ *rand.Rand) {
+		weightMsgSetUserPreferences = DefaultWeightMsgSetUserPreferences
+	})
+
+	return simulation.WeightedOperations{
+		simulation.NewWeightedOperation(weightMsgJoinService, SimulateMsgJoinService(ak, bk, opk, sk)),
+		simulation.NewWeightedOperation(weightMsgLeaveService, SimulateMsgLeaveService(ak, bk, opk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgAddOperatorToAllowList, SimulateMsgAddOperatorToAllowList(ak, bk, opk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgRemoveOperatorhFromAllowList, SimulateMsgRemoveOperatorFromAllowlist(ak, bk, opk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgBorrowPoolSecurity, SimulateMsgBorrowPoolSecurity(ak, bk, pk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgCeasePoolSecurityBorrow, SimulateMsgCeasePoolSecurityBorrow(ak, bk, pk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgDelegatePool, SimulateMsgDelegatePool(ak, bk, k)),
+		simulation.NewWeightedOperation(weightMsgDelegateOperator, SimulateMsgDelegateOperator(ak, bk, opk, k)),
+		simulation.NewWeightedOperation(weightMsgDelegateService, SimulateMsgDelegateService(ak, bk, sk, k)),
+		simulation.NewWeightedOperation(weightMsgSetUserPreferences, SimulateMsgSetUserPreferences(ak, bk, sk, k)),
+	}
+}
+
+func SimulateMsgJoinService(
+	ak authkeeper.AccountKeeper,
+	bk bankkeeper.Keeper,
+	opk types.OperatorsKeeper,
+	sk types.ServicesKeeper,
+) simtypes.Operation {
+	return func(
+		r *rand.Rand, app *baseapp.BaseApp, ctx sdk.Context, accs []simtypes.Account, chainID string,
+	) (simtypes.OperationMsg, []simtypes.FutureOperation, error) {
+		msg := &types.MsgJoinService{}
+
+		operatorsKeeper := opk.(*operatorskeeper.Keeper)
+		servicesKeeper := sk.(*serviceskeeper.Keeper)
```

Comment:
What's the point of using interfaces as the params type, if you are going to cast them anyway? I think it's better to either:

1. Require non-interface types directly as params, or
2. Use interfaces everywhere

This way you don't have to downcast anything and it's easier to manage everything